src/modules/carrierroute/cr_db.c

```diff
@@ -352,6 +353,7 @@ int load_route_data_db(struct route_data_t * rd) {
 		}
 	}
 	int n = 0;
+	boolean query_done = true;
```

Comment:
Please change query_done to false here and adapt the code accordingly. The name of the variable should reflect the value it holds.